File: kernel/src/main/java/com/sos/scheduler/engine/kernel/plugin/PlugInFactory.java
package com.sos.scheduler.engine.kernel.plugin;

import com.sos.scheduler.engine.kernel.Scheduler;
import org.w3c.dom.Element;

/** Ein Plugin muss eine statische Methode implementieren, die Objekt dieser Klasse liefert.
 * Die Methode wird über Reflection aufgerufen.
 * 
 * public static PlugInFactory newFactory();
*/
public interface PlugInFactory {
    PlugIn newInstance(Scheduler scheduler, Element configurationOrNull);
}

File: playground/zschimmer/src/main/scala/com/sos/scheduler/engine/playground/zschimmer/PluginFactory.scala
package com.sos.scheduler.engine.playground.zschimmer

import com.sos.scheduler.engine.kernel.Scheduler
import com.sos.scheduler.engine.kernel.plugin.Plugin
import com.sos.scheduler.engine.kernel.plugin.{PluginFactory => JavaPlugInFactory}
import org.w3c.dom.Element
import scala.xml.Elem


trait PlugInFactory extends JavaPlugInFactory {
    def newInstance(scheduler: Scheduler, configurationOption: Option[Elem]): Plugin

    override def newInstance(scheduler: Scheduler, configurationOrNull: Element) = {
        val elem = Option(configurationOrNull) map { XMLs.fromJavaDom(_).asInstanceOf[xml.Elem] }
        newInstance(scheduler, elem)
    }
}
